diff -uNr linux-6.8.9-original/arch/riscv/Kconfig linux-6.8.9/arch/riscv/Kconfig
--- linux-6.8.9-original/arch/riscv/Kconfig	2024-05-02 07:35:35.000000000 -0700
+++ linux-6.8.9/arch/riscv/Kconfig	2024-05-18 23:00:11.477903132 -0700
@@ -625,7 +625,7 @@
 
 config TOOLCHAIN_HAS_ZIHINTPAUSE
 	bool
-	default y
+	default n
 	depends on !64BIT || $(cc-option,-mabi=lp64 -march=rv64ima_zihintpause)
 	depends on !32BIT || $(cc-option,-mabi=ilp32 -march=rv32ima_zihintpause)
 	depends on LLD_VERSION >= 150000 || LD_VERSION >= 23600
